prepare-ocp4-aws-config.yml

```diff
@@ -1,56 +1,176 @@
+# 2. pipeline_repo_path : path to pipeline_repo 
+# 3. aws_region: region on which the ocp_cluster needs to be deployed
+# 4. pipeline_repo_url: url from which the cvp pipeline needs to be cloned 
+#                       if the pipeline_repo_path is empty
+# 5. openshift_install: Boolean value to specify whether to install openshift 
+#                       with generated config
+# 6. openshift_installer_log_level: Logging level of openshift installer
+#                                   defaults to info
+# 7. ocp_install_directory: Directory in which the cluster config files are generated
+#                           by default it is fetched from pipleline_repo_path
+# 8. vault_file: path to vault_file where passwords are stored
+# 9. ocp_install_config_template: ocp_config file template path
+# 10. ocp_installer_url: url to tar file of ocp_installer with ocp version appended to it
+
+# pipeline repo path is directory where the cvp pipeline repo exists 
+# ie., https://gitlab.sat.engineering.redhat.com/cvp/pipeline.git
+
+# vault file is needed for decrypting credentials for openshift run
+# vault file is inside the pipeline_repo_path
+
+# All we need is pipeline_repo_path to be intialized 
+# if the folder is empty, the playbook clones the contents of the pipeline repo url
+
+- name: "Prepare OCP 4.x configuration for installation on AWS"
+  hosts: localhost
   become: false
   gather_facts: false
+  pre_tasks:
+    # set the default variable using set_fact
+    - name: "Set default varibles such that they can be overridden by extra-vars and environment variables"
+      set_fact:
+        cluster_name: "{{ cluster_name | default('test-cluster') }}"
+        aws_region: "{{ aws_region | default('us-east-2') }}"
+        pipeline_repo_path: "{{ pipeline_repo_path | default('') }}"
+        pipeline_repo_url: "https://gitlab.sat.engineering.redhat.com/cvp/pipeline.git"
+        openshift_install: "{{ openshift_install | default(false) }}"
+        openshift_installer_log_level: "{{ openshift_installer_log_level | default('info') }}"
 
-  vars_files:
-    - "{{ playbook_dir }}/../../../config/ansible/vault.yml"
+    - name: "Set OCP install directory & vault_file path"
+      # Note: all the set_fact variables are fetched from extra-vars passed to playbook
+      # if the variables are not found they would be assuming the default variables
+      set_fact:
+        ocp_install_directory: "{{ ocp_install_directory | 
+                                   default(pipeline_repo_path+'/operators/ocp-cluster-deployment/') }}"
+        vault_file: "{{ vault_file | 
+                        default(pipeline_repo_path+'/config/ansible/vault.yml')}}"
+        ocp_install_config_template: "{{ ocp_install_config_template | 
+                                         default(pipeline_repo_path+'/operators/ocp-cluster-deployment/install-config.yaml.j2') }}"
+        ocp_installer_url:  "{{ ocp_installer_url | 
```

Comment:
Such URL does not exist.  HTTP 404. Because they always change version in the sable directory.
And default should work for people. Otherwise they would be confused.

Reply:
ehh, next line :-)